xds/src/main/java/io/grpc/xds/client/XdsClient.java

```diff
@@ -407,26 +399,35 @@ void handleResourceResponse(
 
     /** Called when the ADS stream is closed passively. */
     // Must be synchronized.
-    void handleStreamClosed(Status error);
+    void handleStreamClosed(Status error, boolean inRetry, ControlPlaneClient cpcThatClosed);
```

Comment:
You're creating a new `XdsResponseHandler` for each `ControlPlaneClient`, and passing it the `ServerInfo`. Why are we also passing the `ControlPlaneClient` to the handler? I'd expect only one or the other: use a shared handler and pass it the ControlPlaneClient in each method, or one handler per client and don't pass the client.

Reply:
Good point.  Stopped sending the ControlPlaneClient and just look it up.